### PR TITLE
Make to_device method public

### PR DESCRIPTION
**Pull request by unnonouno**
I made `to_device` method public for users who want to make a custom converter method for `Updater`. I sometimes copy this method to my code by hand.

### CODE REVIEW

**Comment by okuta** [comment]
Will you write document string about `to_device`?

**Comment by unnonouno** [comment]
OK. I also updated the document.

**Comment by okuta** [comment]
LGTM except test.
Please add test case.

File: tests/chainer_tests/dataset_tests/test_convert.py [hunk 1]
```diff
@@ -210,4 +210,67 @@ def test_concat_dicts_padding_gpu(self):
...
+
+    @attr.gpu
+    def test_to_device(self):
+        src_xp = get_xp(self.src_gpu)
+        dst_xp = get_xp(self.dst_gpu)
+        x = src_xp.array([1], 'i')
+        y = dataset.to_device(self.device, x)
+        self.assertIsInstance(y, dst_xp.ndarray)
+
+        if self.device >= 0:
```
Comment (okuta) [inline tests/chainer_tests/dataset_tests/test_convert.py]: This test is failed on Python3. Please fix.
```
======================================================================
ERROR: test_to_device (test_convert.TestToDeviceGPU_param_0)  parameter: {'dst_gpu': True, 'src_gpu': True, 'device': None}
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work/chainer/tests/chainer_tests/dataset_tests/test_convert.py", line 252, in test_to_device
    if self.device >= 0:
TypeError: unorderable types: NoneType() >= int()
```

**Comment by unnonouno** [comment]
I fixed it

**Comment by okuta** [comment]
LGTM!

**Pull request merged**